### DISCUSSION TITLE
What is the difference between 2 types of xgboost usage?

### DISCUSSION THREAD

**Discussion by bilative** [aws/sagemaker-python-sdk#2397] [Help]
Hi,

I have a question like this:
What is the difference between 
```python
from sagemaker.amazon.amazon_estimator import get_image_uri
container = get_image_uri(region, 'xgboost', '1.0-1')
```

and this

```python
from sagemaker.xgboost.estimator import XGBoost
estimator = XGBoost(entry_point = "your_xgboost_abalone_script.py", 
                    framework_version='1.0-1',
                    hyperparameters=hyperparameters,
                    role=sagemaker.get_execution_role(),
                    train_instance_count=1,
                    train_instance_type='ml.m5.2xlarge',
                    output_path=output_path)
```

What is the 2 type of usage difference?
Actually i mostly used sklearn until nowadays. And Sagemaker's logic little push me hard.
Thanks in advance.

**Comment by ajaykarpur** [comment]
Hi @bilative, which version of the SageMaker Python SDK are you using? You can check the version with `sagemaker.__version__`

---

The `get_image_uri` functionality (as well as `image_uris.retrieve` in v2) is a utility that's used to create the container URI. This functionality is called by the `XGBoost` class to construct the container URI.

Instead of calling `get_image_uri`, you can simply specify the `framework_version` as you did in the second code snippet. There is no need to create the URI yourself, as in the first snippet.

**Comment by bilative** [comment]
@ajaykarpur Thanks for the suggestion.
I'll think about this.

I checked my version and i saw that "1.72.1" After you said this to me, i realised on the [built in xgboost websize](https://docs.aws.amazon.com/sagemaker/latest/dg/xgboost.html) SDK1 and SDK2 options of the usage.

Do you know anything about, are there any speed difference between uri version,  SDK1 version or SDK2 version. Or there is no difference except usage?

**Comment by ajaykarpur** [comment]
Sure thing. You may also want to take a look at these two documentation pages:
- https://sagemaker.readthedocs.io/en/stable/frameworks/xgboost/using_xgboost.html
- https://sagemaker.readthedocs.io/en/stable/v2.html

> Do you know anything about, are there any speed difference between uri version, SDK1 version or SDK2 version. Or there is no difference except usage?

The only differences are related to usage, and these differences are highlighted in the above documentation page. We recommend upgrading to SDK v2. You can do so in the following manner:
1. Upgrade the installed SDK version with `pip install --upgrade sagemaker`
2. Use the [automatic upgrade tool](https://sagemaker.readthedocs.io/en/stable/v2.html#automatically-upgrade-your-code) to automatically modify your existing code to work with SDK v2.
